### DISCUSSION TITLE
.send() not passing message to the system?

### DISCUSSION THREAD

**Discussion by angusturner** [mido/mido#534] [Q&A]
I am trying to send note information to the system midi port, in order to have it synthesised in real-time by garageband. 

I can receive keyboard input fine, using `.recieve()` or `.poll()` etc., but when I send note events to the same port the system does not seem to detect them? 

I have tried using both`.open_ioport()` and `.open_output()`.

I am on Mac OS X 10.11, using the rtmidi backend and Python 3.6.

**Comment by olemb** [comment]
I'm not sure what you mean by system MIDI port. On my system there is are
no MIDI ports out of the box. What name do you use when you open it?

What I've found to work best is to set up an IAC driver bus:


https://hearandknow.wordpress.com/2010/05/09/iac-get-virtual-midi-ports-on-a-mac/

You can then open the port using the name you chose, for example:

    port = mido.open_output('Bus 1')

Any messages you send on this port will picked up by GarageBand since it
listens to all input ports.

**Comment by cjappl** [comment]
Hi there,

This example should work with GarageBand, I'm using 10.12, and also Python 3.6.

Here are the steps:

1. Open GarageBand
2. Load an instrument
3. Select that instrument
4. Open python session, type the following commands:

```
import mido
out = mido.open_output('test-midi', virtual=True)
msg = mido.Message('note_on', note=64, channel=1)
out.send(msg)

```
I have been able to get notes to play with GarageBand using this method! GarageBand even automatically picks up the new port once you call "open_output", so no need to select the midi device.
